libraries/node-core-library/src/JsonFile.ts

```diff
@@ -571,7 +571,10 @@ export class JsonFile {
   }
 
   private static _buildJjuParseOptions(options: IJsonFileParseOptions = {}): jju.ParseOptions {
-    const parseOptions: jju.ParseOptions = {};
+    const parseOptions: jju.ParseOptions = {
+      reserved_keys: 'replace'
```

Comment:
JJU's behavior is absolutely bizarre.  I assumed they were trying to protect against this kind of mistake:

![image](https://github.com/microsoft/rushstack/assets/4673363/342bcba1-8800-49d3-8b1a-a8fbbc7d8c53)

The compiler catches the above case, but neither the compiler nor our lint rules catch `[key]` access like this:

![image](https://github.com/microsoft/rushstack/assets/4673363/a3ab2228-843a-4016-9e2f-3032b2e2cac8)

**Approach #<!-- -->1:** A sensible protection would be to delete the `prototype` so that JSON parsed objects don't have this problem. That's what I thought JJU was doing. @iclanton noted that it breaks `ajv`'s object analysis however, which doesn't expect deleted prototypes.

**Approach #<!-- -->2:** What JJU is actually doing:  It deletes the `prototype` but also discards actual keys using reserved names.  This results in inexplicable data loss:

![image](https://github.com/microsoft/rushstack/assets/4673363/82b8a12d-f80d-4a0e-bc49-a234176a1ecc)

**Approach #<!-- -->3:** The `replace` behavior enabled by this PR is to correctly parse keys like `"constructor"`, allowing the them to shadow the `prototype` property of the same name.  Nothing is actually "replaced", it's just shadowed according to normal JavaScript prototype inheritance.

Empirical experimentation shows that Approach #<!-- -->3 is used by `JSON.parse()` as well as these 3 highly popular parsing libraries:

- `json5`
- `yaml`
- `js-yaml`

Therefore despite the pitfalls of JavaScript, I agree that PR #4736 is the best and most conventional way to go.